rc/src/lib.rs

```diff
@@ -0,0 +1,220 @@
+use std::ops;
+use std::cell::Cell;
+
+pub struct MyRc<T> {
+    data: *mut MyBox<T>,
+}
+
+pub struct MyWeak<T> {
+    data: *mut MyBox<T>,
+}
+
+struct MyBox<T> {
+    data: *mut T,
+    strong: Cell<usize>,
+    weak: Cell<usize>
+}
```

Comment:
I had vaguely remembered one of the presentations had used `Cell` with `Rc`, and I was getting compiler errors that went away when I started using `Cell`